src/GenAPI/Microsoft.DotNet.GenAPI/CSharpSyntaxWriter.cs

```diff
@@ -69,18 +68,31 @@ public CSharpSyntaxWriter(string? exceptionMessage)
                 rs = rs.WithExpressionBody(null);
             }
 
-            if (rs.ReturnType.ToString() != "System.Void")
+            string returnType = rs.ReturnType.ToString();
+            if (returnType != "void" && returnType != "System.Void")
             {
-                rs = rs.WithBody(GetThrowNullBody(true));
+                rs = rs.WithBody(GetThrowNullBody());
             }
             else
             {
-                rs = rs.WithBody(GetEmptyBody(true));
+                rs = rs.WithBody(GetEmptyBody());
             }
 
             return rs.WithParameterList(rs.ParameterList.WithTrailingTrivia(SyntaxFactory.Space));
         }
 
+        /// <inheritdoc />
+        public override SyntaxNode? VisitOperatorDeclaration(OperatorDeclarationSyntax node)
+        {
+            // visit subtree first to normalize type names.
+            OperatorDeclarationSyntax? rs = (OperatorDeclarationSyntax?)base.VisitOperatorDeclaration(node);
```

Comment:
Should this be?
```suggestion
            OperatorDeclarationSyntax? rs = base.VisitOperatorDeclaration(node) as OperatorDeclarationSyntax;
```

If it returns `null`, I don't know if the cast would fail always? (I haven't tried casting into a nullable)